app/Main.hs

```diff
@@ -16,23 +16,16 @@ main
  = do
   maybeVersion <- Task.runExceptT JetpackVersion.load
   case maybeVersion of
-    Left err -> print err
-    Right version -> runAndCheckVersion version
-
-runAndCheckVersion :: JetpackVersion.Version -> IO ()
-runAndCheckVersion version =
-  case Version.check version of
-    Just err -> do
-      print err
-      run
-    Nothing -> run
-
-run :: IO ()
-run = do
-  config <- Config.readConfig
-  args@Args {version, watch} <- readArguments
-  if version
-    then Message.info Version.print
-    else if watch
-           then Watcher.watch config args
-           else Builder.build config args
+    Left err -> Message.error err
+    Right jetpackVersion -> do
+      _ <-
```

Comment:
Interesting! I wasn't aware we did this while we paired and didn't know till I reread that we actually had to do this for the effects to happen / things to work.